packages/dds/tree/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -526,14 +527,18 @@ function getMovedEffect(
 	revision: RevisionTag | undefined,
 	id: MoveId,
 	count: number,
-): MarkEffect | undefined {
+): Detach | undefined {
 	const effect = getMoveEffect(moveEffects, CrossFieldTarget.Destination, revision, id, count);
 	assert(effect.length === count, 0x6f3 /* Expected effect to cover entire mark */);
 	const movedEffect = effect.value?.movedEffect;
-	if (movedEffect !== undefined && movedEffect.type === "MoveOut") {
-		moveEffects.moveKey(CrossFieldTarget.Source, movedEffect.revision, movedEffect.id, count);
+	if (movedEffect === undefined) {
+		return undefined;
 	}
 
+	assert(isDetach(movedEffect), "Expected moved effect to be a detach ");
```

Comment:
Can we change the type of the (unfortunately named) field `MoveEffect.movedEffect` to `Detach` so that this assert isn't necessary?

Reply:
Good idea, but tell me more: what makes you say that it is unfortunately named and what would you suggest instead? It would be fine to rename this as part of this PR I think.